docs/binding-controller.md

```diff
@@ -0,0 +1,60 @@
+```mermaid
+%%{init: {'theme':'forest'}}%%
+flowchart TD
+  %% Defining the styles
+    classDef Red fill:#FF9999;
+    classDef Amber fill:#FFDEAD;
+    classDef Green fill:#BDFFA4;
+
+  predicate((PREDICATE: <br>SnapshotEnvironmentBinding<br>is created or updated))
```

Comment:
I think following the [scenario controller diagram](https://github.com/redhat-appstudio/integration-service/blob/main/docs/scenario-controller.md?plain=1#L12) logic would be good here.  The predicate in that chart uses a statement instead of a question

Reply:
Ah, sorry for the misunderstanding. The question mark wasn't part of the suggestion, I was just curious about what you'd think about it. Removed it now.

Reply:
Got it, I'm fine with that change,  I'll make it now